cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go

```diff
@@ -28,15 +28,22 @@ import (
 type InitConfiguration struct {
 	metav1.TypeMeta `json:",inline"`
 
+	// Metadata is the standard object metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	Metadata *metav1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
also not embedding it here, which is not consistent with other core and external types.

question for @vincepri @fabriziopandini :

CAPI is not using the stock ObjectMeta since it had a problem around one of its "time" fields:
https://github.com/kubernetes-sigs/cluster-api/blob/3065a926259f682f65cb8331b5f2543b270882a8/api/v1alpha4/common_types.go#L122-L157

what is your recommendation for kubeadm?
one of the original reasons for the ObjectMeta requests for kubeadm were related to kustomize and the ObjecMeta.Name field, which the CAPI object does not have.

Reply:
Ah thanks, just saw this comment after writing the above. I'd not embed it altogether, is there a reason to have it?

Reply:
frankly, the only requests were from kustomize users for file-on-disk manipulations, which is not a solid enough reason to have the kubeadm types include a partial ObjectMeta or the standard one.

given ObjectMeta is for types that are managed at the apiserver, it feels that we can add it to kubeadm types only once and if they become CRDs.

the discussion on the unmerged KEP brings similar concerns:
https://github.com/kubernetes/enhancements/pull/1816

Reply:
Wondering if someone from API machinery (might be @nikhita) could chime in here; we have two topics at stake
- Should we add ObjectMeta tu kubeadm component config to make them "kustomize friendly" ? 
- If yes, should we add the full ObjectMeta or a subset of it thus avoiding the problems mentioned above ?